### ISSUE TITLE
[M2] Individual Member Task Reflections

### ISSUE THREAD

**Issue by TheRealDexipher** [SIT-P5-7/ICT2201_P5-7#55]
< Each member is required to write no more than 2 paragraphs of what they have done for milestone one, what they have learned, how they were to do it differently if they are given another chance to do it again, and as a team what can be done better. >

**Comment by TheRealDexipher** [comment]
Lee Hui Fang Anne
In M1, time was spent on writing the use case descriptions and identifying the number of transactions there are in the use case description. However, in M2 it took a large amount of time to correct the descriptions as there were some minor changes to the use case model due to the specifications by the client. There were also decisions that needed to be made while estimating the TCF and EF. With the addition of the project effort and the duration estimate, I needed to read up about it online to find out what values are needed to calculate them. Through this, our team has decided to spread out the tasks evenly and within the time constraint, I had to immediately edit the changes from M1 into M2. If I were to turn back time, the tasks should be delegated more evenly and effectively between the members, and started to put together the class diagram, sequence diagram as well as the component diagram. As a team, I feel that we can improve on the overall group efficiency by doing our part with initiative and independently, hence improving on the group productivity.

Dylan Teo
Our team worked closely and everyone had a hand in each pie to try and improve on the report. Aside from being part of the report and diagrams, the main thing I’ve learnt is actually information gathering. I didn’t do enough preparation for the client meeting in a sense that my questions that were asked weren't well thought. There was some undying information that was good to have but it did not strike me at that point in time. Most of the things can be researched and tweaked but not for the lack of information. If given another chance, I would want to properly prepare myself for the client meeting and ask important and relevant questions that will aid me moving forward. Another issue we had was distribution of workload, not that one person was doing too much instead, the efficiency of distribution caused it to be quite messy. This time round our team decided to think about how we wanted to distribute the workload to maximize efficiency instead of waiting on one and another for a certain portion of the work to be done. 

Poh Wee Yee
For M1, I spent most of my time optimizing our use-case diagrams due to the lack of information as we were still finalizing our email to be sent to the client and product scopes which resulted in the slow progress of other parts of the M1 report. This eventually affected the progress of our M2 report as much of the M1 content needed to be adjusted and added which also includes the use-case diagram that affected the progress of our use-case descriptions. Through this, we then decided to spread out the task evenly to cover more content in our M2 report with the strict time constraint we had, I immediately started working on our class and component diagrams as it would affect our M2 report significantly. If I could do it all over again, I would have delegated the task more effectively and started brainstorming for things we needed to confirm with the client so that we could gather that information earlier for important components of the report. As a team, I think we could improve our overall efficiency by taking the initiative to do our part when we can.

Toh Xin Yi 
Upon completing M2, I realise that tasks that were incomplete or undone previously from M1 took up a large amount of time to be corrected during M2. It affected the efficiency of our progress for M2, especially so for the class diagram and sequence diagram. Hence, learning the mistakes made from M1, we had to ensure that we covered all tasks and that they are done accurately in M2. We should have also gathered all our uncertainties and doubts and clarified them swiftly with the professor which would have prevented us from finalising our use case diagram and descriptions later on into M2. In M2, I explored the various architectural designs and gained knowledge on how they benefit different systems. While completing the sequence diagram, it has helped to verify our system flow as I found myself correcting the use case diagrams along the way. As a team, I believe that we can improve on being more initiative and independently take on tasks which would improve overall productivity.
